### PR TITLE
[MM-30042]: CWS - Step 3 of Getting Started shows error when trying to send 10 email addresses

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
This PR adds the ability to check against the free-tier ```userLimit``` and ```total users``` in the system to determine how many invites can be sent out. The send button is disabled when this limit is exceeded.

#### Ticket Link
[MM-30042](https://mattermost.atlassian.net/browse/MM-30042?atlOrigin=eyJpIjoiNzlmNzJmZTA0NmEwNDhkZmE3YmEwYjFiYTVkMjI5YjkiLCJwIjoiaiJ9)

[MM-29958](https://mattermost.atlassian.net/browse/MM-29958?atlOrigin=eyJpIjoiM2M5YmJjZTcxNzMyNGQ0MjhhNGE3MzI3NzljNTQwMzQiLCJwIjoiaiJ9)

#### Screenshot
In this screenshot, it shows that user can only invite 2 more users on their free-tier and consequently the send button is disabled until they remove the extra email. The appropriate messages and error messages guiding the user on these limitations are calculated.
Screenshot 2021-02-01 at 18 38 45

### CODE REVIEW

**Comment by marianunez** [comment]
@AGMETEOR you also need to add the API documentation for this new link here https://github.com/mattermost/mattermost-api-reference to be able to get past the build error. You can see this example https://github.com/mattermost/mattermost-api-reference/pull/587

File: components/next_steps_view/steps/invite_members_step/invite_members_step.tsx [hunk 1]
```diff
@@ -127,8 +147,13 @@ export default class InviteMembersStep extends React.PureComponent<Props, State>
...
         }
 
-        if (value.length > 10) {
-            this.setState({emailError: Utils.localizeMessage('next_steps_view.invite_members_step.tooManyEmails', 'Invitations are limited to 10 email addresses.')});
+        const {subscriptionStats} = this.props;
+
+        if (value.length > subscriptionStats.remaining_seats) {
+            this.setState({emailError: this.props.intl.formatMessage({
+                id: 'next_steps_view.invite_members_step.tooManyEmails',
+                defaultMessage: 'Invitations are limited to {num}'},
```
Comment (marianunez) [inline components/next_steps_view/steps/invite_members_step/invite_members_step.tsx:155]: Same comment as above

File: components/next_steps_view/steps/invite_members_step/invite_members_step.tsx [hunk 2]
```diff
@@ -138,7 +163,16 @@ export default class InviteMembersStep extends React.PureComponent<Props, State>
...
             const newEmails = [...this.state.emails, ...emails];
-            this.setState({emails: newEmails, emailInput: '', emailError: newEmails.length > 10 ? Utils.localizeMessage('next_steps_view.invite_members_step.tooManyEmails', 'Invitations are limited to 10 email addresses.') : undefined});
+            const {subscriptionStats} = this.props;
+
+            this.setState({
+                emails: newEmails,
+                emailInput: '',
+                emailError: newEmails.length > subscriptionStats.remaining_seats ? this.props.intl.formatMessage({
+                    id: 'next_steps_view.invite_members_step.tooManyEmails',
+                    defaultMessage: 'Invitations are limited to {num}'},
```
Comment (marianunez) [inline components/next_steps_view/steps/invite_members_step/invite_members_step.tsx:173]: Another occurrence

File: components/next_steps_view/steps/invite_members_step/invite_members_step.tsx [hunk 3]
```diff
@@ -224,7 +258,10 @@ export default class InviteMembersStep extends React.PureComponent<Props, State>
                             </h3>
                             <FormattedMessage
                                 id='next_steps_view.invite_members_step.youCanInviteUpTo'
-                                defaultMessage='You can invite up to 10 team members using a space or comma between addresses'
+                                defaultMessage='You can invite up to {members} team members using a space or comma between addresses'
```
Comment (marianunez) [inline components/next_steps_view/steps/invite_members_step/invite_members_step.tsx:261]: Here I believe we actually just want to show what is the max they can have in their free tier and not how much they have remaining. Posted a question in the channel to get some input on what is the best message here: https://community-daily.mattermost.com/private-core/pl/ao8ghbr4oiyiimy8eor1eqrtky

File: package.json [hunk 4]
```diff
@@ -39,7 +39,7 @@
     "localforage-observable": "2.0.1",
     "mark.js": "8.11.1",
     "marked": "github:mattermost/marked#87769262aa02e1784570f61f4f962050e07cc335",
-    "mattermost-redux": "github:mattermost/mattermost-redux#26d73f80d7bd4c73e65bc3168058a50cbcae251b",
+    "mattermost-redux": "github:mattermost/mattermost-redux#191efaea2a359933c5bf0279de53448bfe9bd1e3",
```
Comment (marianunez) [inline package.json:42]: This should also be updated in the package-lock.json

**Review by marianunez — CHANGES_REQUESTED** [review]
LGTM overall, some comments on wording of the error

File: components/next_steps_view/steps/invite_members_step/invite_members_step.tsx [hunk 5]
```diff
@@ -107,11 +123,15 @@ export default class InviteMembersStep extends React.PureComponent<Props, State>
...
             const newEmails = [...this.state.emails, ...emails];
+            const {subscriptionStats} = this.props;
 
             this.setState({
                 emails: newEmails,
                 emailInput: '',
-                emailError: newEmails.length > 10 ? Utils.localizeMessage('next_steps_view.invite_members_step.tooManyEmails', 'Invitations are limited to 10 email addresses.') : undefined,
+                emailError: newEmails.length > subscriptionStats.remaining_seats ? this.props.intl.formatMessage({
+                    id: 'next_steps_view.invite_members_step.tooManyEmails',
+                    defaultMessage: 'Invitations are limited to {num}'},
```
[the remark this answers is not in the document]
Comment (marianunez) [reply components/next_steps_view/steps/invite_members_step/invite_members_step.tsx:133]: The wording was update in @pablovelezvidal PR here: https://github.com/mattermost/mattermost-webapp/pull/7446/files#diff-1a490f08504158838befaa816ee8d5c08dff9e694d5a568ce4b0a4398dcc563eL3437-R3437 
This PR is adding the correct number and his PR is adding the correct text :) there is overlap so you may need to coordinate there the conflict resolutions.

File: i18n/es.json [hunk 6]
```diff
@@ -3405,8 +3405,8 @@
   "next_steps_view.invite_members_step.send": "Enviar",
   "next_steps_view.invite_members_step.sendInvitationsViaEmail": "Enviar invitaciones por correo electrónico",
   "next_steps_view.invite_members_step.shareThisLink": "Comparte este enlace para invitar miembros",
-  "next_steps_view.invite_members_step.tooManyEmails": "Las invitaciones están limitadas a 10 direcciones de correo electrónico.",
-  "next_steps_view.invite_members_step.youCanInviteUpTo": "Puede invitar hasta 10 miembros del equipo utilizando un espacio o una coma entre las direcciones",
+  "next_steps_view.invite_members_step.tooManyEmails": "Las invitaciones están limitadas a {num} direcciones de correo electrónico.",
```
Comment (pablovelezvidal) [inline i18n/es.json:3408]: hi @AGMETEOR . I think that we only modify the en.json file when we do changes, but @marianunez can help us to confirm that.

File: components/next_steps_view/steps/invite_members_step/invite_members_step.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (pablovelezvidal) [reply components/next_steps_view/steps/invite_members_step/invite_members_step.tsx:134]: We already coordinated with @AGMETEOR  and I will apply the text modifications on top of the changes in this PR.

**Review by pablovelezvidal — APPROVED** [review]

File: components/next_steps_view/steps/invite_members_step/invite_members_step.tsx (hunk 3 above)
Comment (marianunez) [reply components/next_steps_view/steps/invite_members_step/invite_members_step.tsx:262]: Based on Adam's respond on that thread, this is showing the correct remaining users. Will resolve this comment.

File: i18n/es.json (hunk 6 above)
Comment (marianunez) [reply i18n/es.json:3408]: Yes, the other language files are handled by the community that translates them all when new texts are added or modified.

**Review by marianunez — APPROVED** [review]
LGTM. Thanks @AGMETEOR!

**Review by nickmisasi — APPROVED** [review]
LGTM

**Comment by stevemudie** [comment]
@AGMETEOR I added users via the `/test` command then refreshed the screen and tried to enter 2 email addresses. 
It disabled the send button but still shows "Invitations are limited to 10 email addresses."

It should tell me I can only send 1 right?

image

**Comment by AGMETEOR** [comment]
> @AGMETEOR I added users via the `/test` command then refreshed the screen and tried to enter 2 email addresses.
> It disabled the send button but still shows "Invitations are limited to 10 email addresses."
> 
> It should tell me I can only send 1 right?
> 
> image

Let me consult Maria on this.

**Comment by AGMETEOR** [comment]
> @AGMETEOR I added users via the `/test` command then refreshed the screen and tried to enter 2 email addresses.
> It disabled the send button but still shows "Invitations are limited to 10 email addresses."
> 
> It should tell me I can only send 1 right?
> 
> image

Hi Steve. The number is ok, just needed to update the error message so it makes sense.

**Review by stevemudie — APPROVED** [review]
Tested and passed.

While I initially expected to see the number in the red text error read the same as the actual amount of users left for me to add - in this case 3 - I understand that explaining that it's limited to 10 by design and that worked as expected.

Looks good!

image

**Pull request merged**